src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -352,22 +352,49 @@ export class StudyViewPageStore {
     @action
     updateNAValue = (uniqueKey: string): void => {
         let newValue = this.isShowNAChecked(uniqueKey);
-        let newFilter = _.clone(this._clinicalDataBinFilterSet.get(uniqueKey))!;
-        newFilter.showNA = newValue;
-        this._clinicalDataBinFilterSet.set(uniqueKey, newFilter);
+        if (this.isGenericAssayChart(uniqueKey)) {
+            let newFilter = _.clone(
+                this._genericAssayDataBinFilterSet.get(uniqueKey)
+            )!;
+            newFilter.showNA = newValue;
+            this._genericAssayDataBinFilterSet.set(uniqueKey, newFilter);
+        } else {
+            let newFilter = _.clone(
+                this._clinicalDataBinFilterSet.get(uniqueKey)
+            )!;
+            newFilter.showNA = newValue;
+            this._clinicalDataBinFilterSet.set(uniqueKey, newFilter);
+        }
     };
 
     @action
     toggleNAValue = (uniqueKey: string): void => {
         this.updateNAValue(uniqueKey);
-        let newFilter = _.clone(this._clinicalDataBinFilterSet.get(uniqueKey))!;
-        let showNA = newFilter.showNA;
-        newFilter.showNA = !showNA;
-        this._clinicalDataBinFilterSet.set(uniqueKey, newFilter);
+        if (this.isGenericAssayChart(uniqueKey)) {
+            let newFilter = _.clone(
+                this._genericAssayDataBinFilterSet.get(uniqueKey)
+            )!;
+            let showNA = newFilter.showNA;
+            newFilter.showNA = !showNA;
+            this._genericAssayDataBinFilterSet.set(uniqueKey, newFilter);
+        } else {
+            let newFilter = _.clone(
+                this._clinicalDataBinFilterSet.get(uniqueKey)
+            )!;
+            let showNA = newFilter.showNA;
+            newFilter.showNA = !showNA;
+            this._clinicalDataBinFilterSet.set(uniqueKey, newFilter);
+        }
     };
 
     public isShowNAChecked = (uniqueKey: string): boolean => {
-        let filter = _.clone(this._clinicalDataBinFilterSet.get(uniqueKey));
+        let filter;
+        if (this.isGenericAssayChart(uniqueKey))
+            filter = _.clone(
+                this._genericAssayDataBinFilterSet.get(uniqueKey)
+            )!;
+        else filter = _.clone(this._clinicalDataBinFilterSet.get(uniqueKey));
```

Comment:
Personally I prefer a full brace style, but I am ok with this also.

Reply:
@dianab0 lets just put in the full braces.  want to keep style consistency